### PR TITLE
Test dynamic relationships

### PR DESCRIPTION
**Pull request by barryvdh**
Relationships should be detected based on static analysis, so dynamic properties should not be an issue.
(Test currently failing)

### CODE REVIEW

**Comment by barryvdh** [comment]
@mfn I don't understand this testcase exactly. Why is the Display output empty? It throws an error on the commandline, but I can't seem to fetch it. Otherwise it should show that 1 model is generated, so not sure why we expect it to be empty.

@stefanbauer this replicates your issue. Now just need to fix it ;)

**Comment by barryvdh** [comment]
So this is not ideal, but not sure how to get the actual relation without executing the method. It could be with static analysis (php-parser?) but not sure.

**Comment by barryvdh** [comment]
So best option: Use static analysis to detect the type without even running it
2nd best option: add the relationship using default Model instead of the actual model (which at least gives some phpdocs)
3rd best: catch the error (this PR)

**Comment by barryvdh** [comment]
Although static analysis might be doable because we already search for the relationship definition in the code now..

**Review by mfn — APPROVED** [review]
LGTM, thanks for the test!

----
> Why is the Display output empty? It throws an error on the commandline, but I can't seem to fetch it

Good question 😅 I'll look into it!

File: tests/SnapshotPhpDriver.php [hunk 1]
```diff
@@ -9,7 +9,7 @@ class SnapshotPhpDriver implements Driver
 {
     public function serialize($data): string
     {
-        return $data;
+        return (string) $data;
```
Comment (mfn) [inline tests/SnapshotPhpDriver.php:12]: Btw, are you sure this is required? I forgot to mention this, but I tested your PR without the casts and it worked.

Or maybe was it due to error-case I guess?

Comment (barryvdh) [reply tests/SnapshotPhpDriver.php:12]: Yeah in that case it was null.

**Pull request merged**